manifests/enterprise/install.pp

```diff
@@ -0,0 +1,45 @@
+# Class splunk::enterprise::install
+#
+class splunk::enterprise::install {
+
+  if $facts['kernel'] == 'Linux' or $facts['kernel'] == 'SunOS' {
+    include splunk::enterprise::install::nix
+  }
+
+  $_package_source = $splunk::enterprise::manage_package_source ? {
+    true  => $splunk::enterprise::enterprise_package_src,
+    false => $splunk::enterprise::package_source
+  }
+
+  if $splunk::enterprise::package_provider != undef and $splunk::enterprise::package_provider != 'yum' and $splunk::enterprise::package_provider != 'apt' and $splunk::enterprise::package_provider != 'chocolatey' {
```

Comment:
`if $splunk::enterprise::package_provider != undef` is equal to `if $splunk::enterprise::package_provider`

---

`if $splunk::enterprise::package_provider != 'yum' and $splunk::enterprise::package_provider != 'apt' and $splunk::enterprise::package_provider != 'chocolatey'` should be equal to `if not $splunk::enterprise::package_provider in ['apt', 'chocolatey', 'yum']` (but I didn't test it).

Reply:
I'll try cleaning this up, I have a local centos7 build I'm using yum providers with